kolibri/plugins/learn/assets/src/views/AssessmentWrapper/index.vue

```diff
@@ -267,18 +267,23 @@ oriented data synchronization.
         if (this.mastered) {
           return 1;
         }
-        if (this.pastattempts) {
+        if (this.pastattempts.length) {
+          let calculatedMastery;
           if (this.pastattempts.length > this.attemptsWindowN) {
-            return Math.min(
+            calculatedMastery = Math.min(
               this.pastattempts.slice(0, this.attemptsWindowN).reduce((a, b) => a + b.correct, 0) /
                 this.totalCorrectRequiredM,
               1
             );
+          } else {
+            calculatedMastery = Math.min(
+              this.pastattempts.reduce((a, b) => a + b.correct, 0) / this.totalCorrectRequiredM,
+              1
+            );
           }
-          return Math.min(
-            this.pastattempts.reduce((a, b) => a + b.correct, 0) / this.totalCorrectRequiredM,
-            1
-          );
+          // If there are any attempts at all, set some progress on the exercise
+          // because they have now started the exercise.
+          return Math.max(calculatedMastery, 0.001);
```

Comment:
I did something very similar for exams here:

https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/views/classes/AssignedExamsCards.vue#L64-L80

Maybe we should standardize these 'limit' constants? They have a small user-facing impact on the way progress bars are rendered.

Reply:
Possibly - I would actually like to revisit whether the progress bars are particularly helpful too (vs just status icons).